presto-product-tests/README.md

```diff
@@ -1,125 +1,113 @@
 
 ```
-docker-machine create -d virtualbox --virtualbox-memory 4096 <machine>
-eval $(docker env <machine>)
+presto-product-tests/bin/run_on_docker.sh <profile> -x quarantine,big_query
 ```
 
-#### Configuration profiles
-
-Configuration profiles are stored in `presto-product-tests/etc` diretory. There are two such profiles:
-
-**distributed** - consists of single node (pseudo-distributed) hadoop cluster and multiple node presto cluster
- with one coordinator node and at least one worker node. You can select number of workers by below command:
-
-    cd presto-product-tests/etc/distributed
-    docker-compose scale presto-worker=<N>
-
-**singlenode** - consists of single node (pseudo-distributed) hadoop cluster and one node presto cluster
-
-#### Execution
-
-1. Build Presto project (to have presto-server and product tests compiled).
+Possible profiles are:
+- **distributed** - consists of single node (pseudo-distributed) Hadoop cluster and multiple node Presto cluster
+    with one coordinator node and at least one worker node.
+- **singlenode** - consists of single node (pseudo-distributed) Hadoop cluster and one node Presto cluster
 
-    ```
-    ./mvnw install
-    ```
-
-2. Create and start Presto cluster
+*Docker container logs can be found in ```/tmp/presto_docker_logs```*
 
-    ```
-    cd presto-product-tests/etc/<profile>
-    docker-compose up -d
-    ```
+## Debugging tests
 
-3. Wait for Presto to be ready.
+The following steps explain how you can run the product tests from your IDE as regular TestNG tests:
 
-    To see if all the components are ready you can use below command.
+1. Start the Hadoop container
 
     ```
-    select count(node_id) from system.runtime.nodes where state = 'active';
+    docker-compose -f presto-product-tests/conf/docker/singlenode/docker-compose.yml up -d hadoop-master
     ```
-
+    
+    Tip: In order to display container logs run ```docker-compose -f presto-product-tests/conf/docker/singlenode/docker-compose.yml logs```
+    
+2. Set ```hadoop-master``` host in ```/etc/hosts```  
+
+    - On GNU/Linux: ```<container ip> hadoop-master```. Container IP can be obtained with:
+
+        ```
+        docker inspect $(docker-compose -f presto-product-tests/conf/docker/singlenode/docker-compose.yml ps -q hadoop-master) | grep -i IPAddress
+        ```
+
+    - On Mac OS X: ```<docker machine ip> hadoop-master```. Docker machine IP can be obtained with:
+
+        ```
+        docker-machine ip <machine|default>
+        ```
+    
+3. Create Presto run configuration
+    
+    - Use classpath of module: ```presto-main```
+    - Main class: ```com.facebook.presto.server.PrestoServer```
+    - Working directory: ```presto-product-tests/conf/presto```
+    - VM options:
+        ```
+        -ea
+        -Xmx2G
+        -Dconfig=etc/config.properties
+        -Dlog.levels-file=etc/log.properties
+        -DHADOOP_USER_NAME=hive
+        -Duser.timezone=UTC
+        ```
 4. Run product tests
```

Comment:
Say that all of the methods to start a test below are exclusive. You can run tests either from IntelliJ or from the CLI. As it stands right now it seems that to run a product test you need to go through all the steps. Also say that you would have to start another debug configuration if you want to debug the server. I think we should reorganize this section to be oriented towards different tasks a developer would want to do "I want to debug a test from IntelliJ" or "I want to debug the Presto server".